pyresample/geometry.py

```diff
@@ -28,6 +28,7 @@
 from typing import Optional, Sequence, Union
 
 import numpy as np
+import xarray as xr
```

Comment:
This makes xarray a hard requirement on pyresample which it is not. Plus `DataArray` is already imported in a try/except below.